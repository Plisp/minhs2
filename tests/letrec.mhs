main
	= letrec a = 1;
			b = a;
	in a + b;
