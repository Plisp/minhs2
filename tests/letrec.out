main :: Int
	= letrec a :: [Int] = Cons 1 b;
			b :: [Int] = Cons 2 a;
			c :: [Int] = Cons 3 c;
	in (head c) + (head (tail (tail a)));
